drivers/ethernet/eth_sam_gmac.c

```diff
@@ -75,49 +75,243 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 /* RX descriptors list */
 static struct gmac_desc rx_desc_que0[MAIN_QUEUE_RX_DESC_COUNT]
 	__nocache __aligned(GMAC_DESC_ALIGNMENT);
+#if GMAC_QUEUE_NUM > 1
```

Comment:
I have already done this in #23045, but we should handle all queue number variations here (i.e. `GMAC_QUEUE_NUM` = 3, 5, 6).

Although such a device does not exist at this time (as far as we know, that is), there could be one in the future and the current implementation will cause an insane behaviour in that case).

Reply:
One more thing: some declaration ifdefs use, "**total** queue count" and others use "**priority** queue count." I suppose this is because we only have "priority queue count" for the active.

Once we addressed the define naming issue (https://github.com/zephyrproject-rtos/zephyr/pull/22961/files#r383596823), we should use only one variant- preferably, the total queue count.

Reply:
Add and "Co-authored-by: Stephanos Ioannidis <root@stephanos.io>"